art/project.py

```diff
@@ -191,13 +189,18 @@ def run_all(self, force_rerun=False):
         """
         run_id = get_run_id()
         logger_id = add_logger(EXPERIMENT_LOG_DIR / get_new_log_file_name(run_id))
+        changed_steps = []
```

Comment:
I think this function shouldn't be responsible for collecting changed_steps. I'd create self.changed_steps and make changes inside check_if_must_be_run